File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        h = {}
        for num in nums:
            if num in h:
                h[num] += 1
            else: 
                h[num] = 1
            if h[num] > 1:
                return True
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> d = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int diff = target - num;

            if(d.containsKey(diff)) {
                return new int[] { d.get(diff), i} ;
            }

            d.put(num, i);
        }
        return new int[] {};
    }
}
